yarn-project/p2p/src/services/libp2p/libp2p_service.ts

```diff
@@ -231,6 +235,27 @@ export class LibP2PService<T extends P2PClientType = P2PClientType.Full> extends
     const blockProposalTopic = createTopicString(TopicType.block_proposal, protocolVersion);
     const blockAttestationTopic = createTopicString(TopicType.block_attestation, protocolVersion);
 
+    const preferredPeersEnrs: ENR[] = config.preferredPeers.map(enr => ENR.decodeTxt(enr));
+    const directPeers = (
+      await Promise.all(
+        preferredPeersEnrs.map(async enr => {
+          const address = enr.getLocationMultiaddr('tcp');
+          if (address === undefined) {
+            return undefined;
+          }
```

Comment:
Should we fail loudly if a direct peer is misconfigured and doesn't have a tcp address? Or at least log?

Reply:
Sounds good